app/scripts/controllers/modals/editModal.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .controller('EditModalController', function($scope, APIService, DataService, QuotaService, LimitRangesService, $uibModalInstance, $filter) {
+
+    var limitRangesVersion = APIService.getPreferredVersion('limitranges');
+    var resourceQuotasVersion = APIService.getPreferredVersion('resourcequotas');
+    var appliedClusterResourceQuotasVersion = APIService.getPreferredVersion('appliedclusterresourcequotas');
+
+    var amountAndUnit = $filter('amountAndUnit');
+    var usageWithUnits = $filter('usageWithUnits');
+    var usageValue = $filter('usageValue');
+
+    var allocatedAmountAndUnit = amountAndUnit($scope.currentCapacity);
+
+    $scope.projectName = JSON.parse($scope.pvc).metadata.namespace;
+    $scope.claim = {}
+    $scope.claim.capacity = Number(allocatedAmountAndUnit[0]);
+    $scope.claim.unit = allocatedAmountAndUnit[1];
+    $scope.disableButton = true;
+
+    $scope.currentCapacityUnits = angular.copy($scope.claim);
+
+    $scope.units = [{
+      value: "Mi",
+      label: "MiB"
+    }, {
+      value: "Gi",
+      label: "GiB"
+    }, {
+      value: "Ti",
+      label: "TiB"
+    }, {
+      value: "M",
+      label: "MB"
+    }, {
+      value: "G",
+      label: "GB"
+    }, {
+      value: "T",
+      label: "TB"
+    }];
+
+    $scope.expand = function() {
+      $scope.updatedCapacity = $scope.claim.capacity + $scope.claim.unit;
+      $uibModalInstance.close($scope.updatedCapacity);
+    };
+
+    $scope.cancel = function() {
+      $uibModalInstance.dismiss('cancel');
+    };
+
+    $scope.groupUnits = function(unit) {
+      switch (unit.value) {
```

Comment:
A little subjective, but using a `switch` with passthrough + no `default` is a code-smell.  I wonder if this can be done in a way that is less brittle.  

Two possible solutions:

```JavaScript 
// not everyone likes nested ternaries, but this is fairly straightforward,
// though pardon the es2015 syntax:
const groupUnits = (unit) => {
  const binary = ['Mi','Gi','Ti'];
  const decimal = ['M','G','T']; 
  return _.includes(binary, unit)
    ? 'Binary Units'
    : _.includes(decimal, unit)
      ? 'Decimal Units'
      : null;
}

// can also use an object literal approach:
// again, pardon the es2015 syntax:
// this impl is probably quite a bit more performant, though 
// I doubt performance is super important here:

const groupUnits = (unit) => {
  const binary = 'Binary Units';
  const decimal = 'Decimal Units';
  return ({
    'Mi': binary,
    'Gi': binary,
    'Ti': binary,
    'M': decimal,
    'G': decimal,
    'T': decimal
  })[unit];
}
```

Reply:
I used the first approach, please review!